### ISSUE TITLE
expose isSearchStalled in all connectors & widgets

### ISSUE THREAD

**Issue by intrepidws** [algolia/vue-instantsearch#710]
@Haroenv asked me to open this issue from my comment on https://github.com/algolia/vue-instantsearch/issues/317#issuecomment-525480594.

I'm looking for some sort of "ready" or "loaded" event that I can latch on to, so that I can perform an action. In my specific case, I would be using this to trigger the jQuery Match Height plugin. This needs to be done after all of the Vue InstantSearch functionality is completed, as it requires the DOM heights to be set. Another similar use case would be for masonry.

While v1 didn't have this feature either, I was able to sort of back into it (using the solutions mentioned in issue #317) - but it was a hack and that same solution won't work in v2. I think that this type of event would be a good thing to add to the Vue InstantSearch functionality.

**Comment by Haroenv** [comment]
I think the correct way here would be using a custom widget, using `connectHits`. You can then `watch` e.g. `state.hits`. There's a guide available on custom widgets here: https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-own-widgets/vue/

```vue
<template>
  <div v-if="state">
    <ul>
      <li v-for="hit in state.hits" :key="hit.objectID">
        <ais-highlight :hit="hit" attribute="name"/>
      </li>
    </ul>
  </div>
</template>

<script>
import { createWidgetMixin } from 'vue-instantsearch';
import { connectHits } from 'instantsearch.js/es/connectors';

export default {
  mixins: [
    createWidgetMixin({ connector: connectHits })
  ],
  watch: {
    'state.hits'(newHits, oldHits) {
      console.log(newHits, oldHits);
      // do your update here
    }
  }
};
</script>
```

https://codesandbox.io/s/vue-instantsearch-app-smokk

**Comment by Haroenv** [comment]
Specifically for match heights, I'd advice using flex box or css grid however (this is mentioned in the Readme of jquery-match-height too https://github.com/liabru/jquery-match-height#modern-browsers)

**Comment by alexcroox** [comment]
how do you differentiate between no search/filters results and loading? I need to show a loading indicator for those using slow 3G before the results are shown.

**Comment by Haroenv** [comment]
@alexcroox for that you can probably use the "searchStalled" property on connectHits

**Comment by alexcroox** [comment]
@Haroenv that worked great, thank you! Feature suggestion, expose the `searchStalled` on `AisInfiniteHits` etc so we don't need a custom widget :)

**Comment by Haroenv** [comment]
status is available in ais-state-results since #1151 :)
